### PR TITLE
Add bre log improvements

### LINKED ISSUE #2168: Bre log improvements
- For services, bre log should contain bre stack trace, not the ballerina stack trace
- Main should direct all un handled exceptions to bre log

### PR DESCRIPTION
**Pull request by isudana**
Fixes #2168

### CODE REVIEW

File: modules/ballerina-core/src/main/java/org/ballerinalang/runtime/ServerConnectorMessageHandler.java [hunk 1]
```diff
@@ -120,9 +120,10 @@ public static void handleErrorInboundPath(CarbonMessage cMsg, CarbonCallback cal
         String errorMsg = ErrorHandlerUtils.getErrorMessage(throwable);
         String stacktrace = ErrorHandlerUtils.getServiceStackTrace(balContext, throwable);
         String errorWithTrace = errorMsg + "\n" + stacktrace;
-        log.error(errorWithTrace);
         outStream.println(errorWithTrace);
 
+        // bre log should contain bre stack trace, not the ballerina stack trace
+        breLog.error("error : " + errorMsg + ", ballerina service stack trace : " + stacktrace, throwable);
```
Comment (imesh) [inline modules/ballerina-core/src/main/java/org/ballerinalang/runtime/ServerConnectorMessageHandler.java]: Ideally a space is not needed before the colon:
https://www.grammarly.com/handbook/punctuation/colon/

File: modules/ballerina-core/src/main/java/org/ballerinalang/runtime/ServerConnectorMessageHandler.java [hunk 2]
```diff
@@ -141,9 +142,11 @@ public static void handleErrorFromOutbound(Context balContext, Throwable throwab
...
```
Comment (imesh) [inline modules/ballerina-core/src/main/java/org/ballerinalang/runtime/ServerConnectorMessageHandler.java]: Same as above, might not need to keep a space before the colon:
https://www.grammarly.com/handbook/punctuation/colon/

File: modules/ballerina-core/src/main/java/org/ballerinalang/runtime/ServerConnectorMessageHandler.java (hunk 1 above)
Comment (isudana) [inline modules/ballerina-core/src/main/java/org/ballerinalang/runtime/ServerConnectorMessageHandler.java]: Thanks Imesh, will remove the space.

Comment (imesh) [inline modules/ballerina-core/src/main/java/org/ballerinalang/runtime/ServerConnectorMessageHandler.java]: Great! Thanks Isuru!

**Pull request merged**